components/pd_client/src/util.rs

```diff
@@ -74,10 +79,11 @@ impl Stream for HeartbeatReceiver {
             }
             if receiver.is_some() {
                 debug!("heartbeat receiver is refreshed");
+                drop(inner);
```

Comment:
Why?

Reply:
With `inner` locked, `self` is borrowed, then the next `self.receiver = receiver` won't compile.

Reply:
`inner` borrows `self.inner`, next line updates `self.receiver`, they can be borrowed at the same time.

Reply:
Makes sense... But if this line is commentted out, the compiler complains:

```
   |
74 |             let mut inner = self.inner.wl();
   |                             ---- immutable borrow occurs here
...
82 |                 self.receiver = receiver.map(Stream01CompatExt::compat);
   |                 ^^^^ mutable borrow occurs here
...
87 |         }
   |         - immutable borrow might be used here, when `inner` is dropped and runs the `Drop` code for type `std::sync::RwLockWriteGuard`
```

It looks like a borrow checker bug...